modules/ROOT/pages/errors/all-errors.adoc

```diff
@@ -6,6 +6,12 @@
 
 This page lists the current Neo4j error codes, which which are returned along side xref:errors/gql-errors/index.adoc[GQLSTATUS error codes], grouped by type.
 
+[NOTE]
+====
+Starting from 2025.04, the messages of the Neo4j error codes are deprecated and will be removed in a future release.
```

Comment:
There will still be an error message but it is kind of the stability of it we are deprecating. So we will have old stable error messages during the deprecation period and later replace them with new unstable ones which are only for human consumption and cannot be relied on programatically

Reply:
But isn't it the same with every deprecation? All deprecated features are stable until they are removed.

Reply:
When they go to the GQLSTATUS error codes for further information, the first thing they will see is the warning:

Please note that while GQLSTATUS codes remain stable (any changes to them will be breaking), changes to status descriptions associated with these codes are not breaking and may happen at any time. For this reason, parsing the status descriptions or incorporating them into scripts is not recommended.

Reply:
Yes, but the note above can be interpreted as there will be no more error message after the deprecation period which is not the case. The new error message will have the format `42N11: msgPartOfStatusDescription` so they are related to status description but not the same thing. Also I would not call it "the messages of the Neo4j error codes", it is not really the  error code which have a message, but an error/exception

Reply:
Maybe something more similar to the below. (please rephrase it, but this is the gist of it):

Starting from 2025.04, the current error messages of all Neo4j errors are deprecated. In a future release, new error messages will be introduce. Just like for status descriptions, changes to these new error messages will not breaking and may happen at any time. For this reason, parsing the error messages or incorporating them into scripts is not recommended.

Reply:
How about:

Starting from 2025.04, the current error messages are deprecated, and the newly added error messages contain the GQLSTATUS code they are associated with.
The new error messages are not intended to be parsed or used in scripts.
Just like for GQLSTATUS status descriptions, changes to these new error messages are not breaking and may happen anytime.
The GQLSTATUS error codes are the only stable identifiers for errors in Neo4j.
For details, see xref:errors/gql-errors/index.adoc[GQLSTATUS error codes].

Reply:
Interesting. It seems that I didn't understand your suggestions in the first place. I thought that these newly added errors would still be messages for the Neo. errors just with the addition of a GQLSTATUS code. Now, I am not so sure what the current and new mean.

Reply:
Maybe we can take a short chat some time today if it easier to explain not in text

Reply:
Starting from 2025.04, using the error messages programmatically is deprecated. Because error messages might change in a future release, please parse or use GQLSTATUS error codes in your application code.
For details, see xref:errors/gql-errors/index.adoc[GQLSTATUS error codes].